hnn_core/tests/test_network.py

```diff
@@ -227,18 +227,29 @@ def test_cell_response(tmpdir):
     spike_gids = [[1, 3], [5, 7]]
     spike_types = [['L2_pyramidal', 'L2_basket'],
                    ['L5_pyramidal', 'L5_basket']]
-    tstart, tstop = 0.1, 98.4
+    tstart, tstop, fs = 0.1, 98.4, 1000.
+    sim_times = np.arange(tstart, tstop, 1 / fs)
     gid_ranges = {'L2_pyramidal': range(1, 2), 'L2_basket': range(3, 4),
                   'L5_pyramidal': range(5, 6), 'L5_basket': range(7, 8)}
     cell_response = CellResponse(spike_times=spike_times,
                                  spike_gids=spike_gids,
-                                 spike_types=spike_types)
+                                 spike_types=spike_types,
+                                 times=sim_times)
     cell_response.plot_spikes_hist(show=False)
     cell_response.write(tmpdir.join('spk_%d.txt'))
     assert cell_response == read_spikes(tmpdir.join('spk_*.txt'))
 
     assert ("CellResponse | 2 simulation trials" in repr(cell_response))
 
+    cell_response.reset()
+    assert ("CellResponse | 0 simulation trials" in repr(cell_response))
+    # reset clears all recorded variables, but leaves simulation time intact
+    assert len(cell_response.times) == len(sim_times)
+    attributes = ['_spike_times', '_spike_gids', '_spike_types',
+                  '_vsoma', '_isoma']
```

Comment:
this test is problematic because it will still pass if a new attribute is added that is memory heavy. Maybe we should use the size of the object (like @blakecaldwell did in his tests) and assert that it's less than x kB ?

Reply:
I like the idea, but I'm not sure how reliable `sys.getsizeof` is at revealing the true size of a Python object. Here we're talking about attributes that contain lists, which can in principle contain arbitrary objects. What if I make this test fail if `cell_responses` has any other attribute?

Reply:
I think we use `sys.getsizeof` to display the size of `epochs` object in MNE (or is it something else?) But I think it's also fine to just make the test fail if `cell_response` has any other attribute so developer is careful when adding new attributes

Reply:
Ah, didn't see this by read your mind (see 7f0639f)! Developer warning added.